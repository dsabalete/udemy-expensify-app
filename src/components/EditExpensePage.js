import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';
import Modal from 'react-modal';

export class EditExpensePage extends React.Component {
    state = {
        modalIsOpen: false
    };
    onSubmit = (expense) => {
        this.props.startEditExpense(this.props.expense.id, expense);
        this.props.history.push('/');
    };
    onRemove = () => {
        this.setState({ modalIsOpen: true });
    };
    handleCloseModal = () => {
        this.setState({ modalIsOpen: false });
    };
    handleRemove = () => {
        this.handleCloseModal();
        // this.props.startRemoveExpense({ id: this.props.expense.id });
        // this.props.history.push('/');
    };
    handleCancelRemove = () => {
        this.handleCloseModal();
    };
    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Edit Expense</h1>
                    </div>
                </div>
                <div className="content-container">
                    <ExpenseForm
                        expense={this.props.expense}
                        onSubmit={this.onSubmit}
                    />
                    <button
                        className="button button--secondary"
                        onClick={this.onRemove}>Remove Expense</button>
                </div>
                <Modal
                    isOpen={this.state.modalIsOpen}
                    onRequestClose={this.handleCloseModal}
                    ariaHideApp={false}
                >
                    <button onClick={this.handleCloseModal}>Close</button>
                    <div>
                        <p>Are you sure you want to remove it?</p>
                        <button onClick={this.handleRemove}>Yes</button>
                        <button onClick={this.handleCancelRemove}>No</button>
                    </div>
                </Modal>
            </div>
        );
    }
}

const mapStateToProps = (state, props) => ({
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch, props) => ({
    startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
    startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);