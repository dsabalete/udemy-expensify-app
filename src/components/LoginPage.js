import React from 'react';
import { connect } from 'react-redux';
import {
    startLoginGoogle,
    startLoginGithub,
    startLoginFacebook
} from '../actions/auth';

export const LoginPage = ({
    startLoginGoogle,
    startLoginGithub,
    startLoginFacebook
}) => (
        <div className="box-layout">
            <div className="box-layout__box">
                <h1 className="box-layout__title">Expensify</h1>
                <p>It's time to get your expenses under control.</p>
                <button
                    id="buttonGoogle"
                    className="button"
                    onClick={startLoginGoogle}>
                    Login with <span className="google--icon"></span>
                </button>
                <button
                    id="buttonGithub"
                    className="button"
                    onClick={startLoginGithub}>
                    Login with <span className="github--icon"></span>
                </button>
                <button
                    id="buttonFacebook"
                    className="button"
                    onClick={startLoginFacebook}>
                    Login with <span className="facebook--icon"></span>
                </button>
            </div>
        </div>
    );

const mapDispatchToProps = (dispatch) => ({
    startLoginGoogle: () => dispatch(startLoginGoogle()),
    startLoginGithub: () => dispatch(startLoginGithub()),
    startLoginFacebook: () => dispatch(startLoginFacebook()),
});

export default connect(undefined, mapDispatchToProps)(LoginPage);