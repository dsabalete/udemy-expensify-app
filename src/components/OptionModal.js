import React from 'react';
import Modal from 'react-modal';

export const OptionModal = (props) => (
    <Modal
        isOpen={props.isOpen}
        onRequestClose={props.onRequestClose}
        ariaHideApp={false}
        className="modal"
        >
        <div>
            <h3>Are you sure you want to remove it?</h3>
            <button className="button" onClick={props.handleRemove}>Yes</button>
            <button className="button" onClick={props.handleCancelRemove}>No</button>
        </div>
    </Modal>
);

export default OptionModal;