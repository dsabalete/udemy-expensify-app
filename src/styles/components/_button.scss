.button {
    background: $blue;
    border: none;
    color: white;
    display: inline-block;
    font-size: $font-size-large;
    font-weight: 300;
    line-height: 1;
    padding: $s-size;
    text-decoration: none;
}

.google--icon {
    background: url('/images/google-logo.png') transparent 5px 50% no-repeat;
    background-size: 32px;
    display: inline-block;
    vertical-align: middle;
    width: 42px;
    height: 42px;
}

.github--icon {
    background: url('/images/github-logo.png') transparent 5px 50% no-repeat;
    background-size: 32px;
    display: inline-block;
    vertical-align: middle;
    width: 42px;
    height: 42px;
}

.facebook--icon {
    background: url('/images/facebook-logo.png') transparent 5px 50% no-repeat;
    background-size: 32px;
    display: inline-block;
    vertical-align: middle;
    width: 42px;
    height: 42px;
}

.button--link {
    background: none;
    border: none;
    color: white;
}

.button--secondary {
    background: $soft-grey;
}