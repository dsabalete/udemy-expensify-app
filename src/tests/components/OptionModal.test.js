import React from 'react';
import { shallow } from 'enzyme';
import { OptionModal } from '../../components/OptionModal';

let onRequestClose, history, wrapper;

beforeEach(() => {
    onRequestClose = jest.fn();
    history = { push: jest.fn() };
    wrapper = shallow(<OptionModal onRequestClose={onRequestClose}/>);
});

test('should render OptionModal', () => {
    expect(wrapper).toMatchSnapshot();
});