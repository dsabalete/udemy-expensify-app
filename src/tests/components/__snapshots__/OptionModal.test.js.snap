// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render OptionModal 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={0}
  isOpen={false}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <div>
    <h3>
      Are you sure you want to remove it?
    </h3>
    <button
      className="button"
    >
      Yes
    </button>
    <button
      className="button"
    >
      No
    </button>
  </div>
</Modal>
`;
